pkg/cli/search_cmd.go

```diff
@@ -0,0 +1,148 @@
+//go:build search
+// +build search
+
+package cli
+
+import (
+	"os"
+	"path"
+
+	"github.com/briandowns/spinner"
+	"github.com/spf13/cobra"
+
+	zotErrors "zotregistry.io/zot/errors"
+)
+
+//nolint:dupl
+func NewSearchCommand(searchService SearchService) *cobra.Command {
+	searchImageParams := make(map[string]*string)
+
+	var servURL, user, outputFormat string
+
+	var isSpinner, verifyTLS, verbose, debug bool
+
+	imageCmd := &cobra.Command{
+		Use:   "search [config-name]",
+		Short: "Search images and their tags",
+		Long: `Search repos or images
+Example:
+  # For repo search specify a substring of the repo name without the tag
+  zli search --query test/repo
+
+  # For image search specify the full repo name followed by the tag or a prefix of the tag.
+  zli search --query test/repo:2.1.
```

Comment:
Maybe we need an example for the syntax used for search by subject.

Reply:
Yes pls